components/skus/browser/skus_utils.cc

```diff
@@ -59,8 +60,21 @@ std::string GetEnvironmentForDomain(const std::string& domain) {
   return "";
 }
 
-void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
+void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   registry->RegisterDictionaryPref(prefs::kSkusState);
 }
 
+void RegisterProfilePrefsForMigration(PrefRegistrySimple* registry) {
+  registry->RegisterDictionaryPref(prefs::kSkusState);
+}
+
+void MigrateSkusSettings(PrefService* profile_prefs, PrefService* local_prefs) {
+  auto* obsolete_pref = profile_prefs->Get(prefs::kSkusState);
+  if (!obsolete_pref || !obsolete_pref->is_dict())
+    return;
+
+  local_prefs->Set(prefs::kSkusState, obsolete_pref->Clone());
+  profile_prefs->ClearPref(prefs::kSkusState);
```

Comment:
I'm not sure why, but when migration happened the state was lost in SKU state

## Before (in `Default/Preferences`)
![image](https://user-images.githubusercontent.com/4733304/191134283-3597d2b9-460b-4f0f-8336-c43810b8dac5.png)


## After (in `Local State`)
![image](https://user-images.githubusercontent.com/4733304/191134213-517b3a42-b006-45ef-acb1-0f0952c3d2fe.png)

Reply:
The rest of the prefs moved over just fine though 😄👍

Reply:
I'm not seeing this code run until exit actually. That might be contributing to the problem as (during my test) I made a new profile to test if the `local_state` was picked up (which it wasn't)

Reply:
I think it should be early return when local state's `kSkusState` is not the default value.
and maybe vpn prefs also has same issue?

Reply:
I think we need to store a boolean for if migration has been done (stored in local state). Something like:
```
// Added 9/2022
registry->RegisterBooleanPref(kBraveVpnLocalStateMigration, false);
```

If `brave_vpn::prefs::kBraveVPNRootPref` in local state has a value, we can set this to true and avoid calling `MigrateSkusSettings` again

Reply:
Added migration guards to both  Skus and VPN to be able safely add profile preferences in future